### PR TITLE
[WordPress → Jetpack app branding] Update Get Apps section

### PR DESCRIPTION
**Pull request by SiobhyB**
_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

This PR update references to the mobile app found on the Get Apps screen (found via _Me_ → _Get Apps_ in Calypso). The design has also be refreshed. 

<details>
 <summary>Here's a high-level overview of the code changes ⤵️</summary>

* The `jetpack/app-branding` feature flag has been used to limit the changes to Calypso's staging site, for now. Note, in the future, this feature flag will be removed along with all Jetpack-specific class names and WordPress-app-specific code. 
* `me/sidebar/index.jsx`: The sidebar has been updated to use the Jetpack icon from the [list of available Gridicons](http://automattic.github.io/gridicons/).
* `get-apps/index.jsx`: A body class has been added to enable the setting of a white background, plus the removal of extra default padding and other styles that don't match our desired design.
* For setting the two different background colours, the same CSS pattern was used as the one found for the WooCommerce landing page (see _My Sites_ → _WooCommerce_ for an example).
* A `getJetpackBrandedPanel()` function was created, returning the newly designed components, in both `get-apps/mobile-download-card.jsx` and `get-apps/desktop-download-card.jsx`. 
</details>

#### Screenshots

<details>
 <summary>Before/After (Desktop) ⤵️</summary>

Screenshots taken on a MacBook Pro using Google Chrome:

| Before  | After |
| ------------- | ------------- |
|  |  |

</details>

<details>
 <summary>Before/After (Mobile) ⤵️</summary>

Screenshots taken from an iPhone XR emulator using Chrome:

| Before  | After |
| ------------- | ------------- |
|  |  |

</details>

#### Testing Instructions

* Navigate to [the auto-generated live link](https://github.com/Automattic/wp-calypso/pull/69674#issuecomment-1299360337) then head to _Me_ → _Apps_. 
* Verify that the design displays as expected across different devices, colour schemes, etc.
* If testing on a development branch, it'd be helpful to verify that the older/current design (with WordPress app references) displays as expected when the `jetpack/app-branding` feature flag is disabled.  

#### Future Iterations

The following additional changes were included in the initial designs. I plan to address these in separate PRs, as they require additional changes:

* Add second Mac button for Apple Silicon.
* Add email input to retrieve a desktop download link from a mobile device.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

**Comment by SiobhyB** [comment]
👋 @paulforgione, I'd appreciate any feedback from a design perspective on this first pass. :) You can see the updates via the live branch link [here](https://github.com/Automattic/wp-calypso/pull/69674#issuecomment-1299360337). I'd appreciate any feedback from a design perspective. Some things I'm not certain about:

* In the desktop app section, I used the same icon across both desktop and mobile devices, though the designs show different icons. Should I change this to use the different version of the WordPress icon for desktop (as per the design)?

* I know there was some discussion related to the mobile app section's copy. Should this be changed?

* The font-sizes don't all precisely match up to the designs. In cases where the font-size in the designs didn't have a matching variable from this list, I went for the closest match:

https://github.com/Automattic/wp-calypso/blob/e054ff394bdf31cbc609e16376a3212a7c889b08/packages/typography/styles/variables.scss#L38-L48

Let me know what you think of the above as well as if there are any other changes that are needed. (Also, as per the "notes" section, there are a couple of other changes in the designs that I plan to work on in future PRs.)

File: client/blocks/get-apps/desktop-download-card.jsx [hunk 1]
```diff
@@ -53,32 +61,62 @@ class DesktopDownloadCard extends Component {
...
 				return {
 					firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
 					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
 					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
+					...( displayJetpackAppBranding && {
+						firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
+						secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
+						thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
+					} ),
 				};
```
Comment (dcalhoun) [inline client/blocks/get-apps/desktop-download-card.jsx:92]: Nit: rather than mutating every property in object, WDYT about returning a different object? IMO it may be little easier to comprehend and, eventually, remove the conditional. If we choose to make this modification, we should apply it to the other object spreads in this function as well. 

```suggestion
				return displayJetpackAppBranding ? {
          firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
          secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
          thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
        } : {
					firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
				};
```

File: client/blocks/get-apps/desktop-download-card.jsx [hunk 2]
```diff
@@ -153,22 +229,122 @@ class DesktopDownloadCard extends Component {
...
+				</div>
+			</>
+		);
+	}
+
+	getJetpackBrandedPanel() {
+		const { isMobile } = userAgent;
+
+		return (
+			<>
```
Comment (dcalhoun) [inline client/blocks/get-apps/desktop-download-card.jsx]: Nit: this Fragment appears unnecessary since there is a single child.

File: client/blocks/get-apps/desktop-download-card.jsx [hunk 3]
```diff
@@ -53,32 +61,62 @@ class DesktopDownloadCard extends Component {
...
+			case MAC_LINK:
+				return <SVGIcon name="apple-logo" size="24" icon={ Apple } />;
+			case 'Linux i686':
+			case 'Linux i686 on x86_64':
+			case LINUX_TAR_LINK:
+			case LINUX_DEB_LINK:
+				return <SVGIcon name="linux-logo" size="24" icon={ Linux } />;
+			case WINDOWS_LINK:
+			default:
+				return <SVGIcon name="windows-logo" size="24" icon={ Windows } />;
```
Comment (dcalhoun) [inline client/blocks/get-apps/desktop-download-card.jsx]: It seems these icons are decorative in their use. WDYT about hiding them from screen readers? Currently, focusing on the Apple image results in merely "image" read aloud. The other icons are only used nested within anchors, so they are not focusable, but it may make sense to hide them for consistency given they do not have `title` attributes. 

For context, CSS-Tricks has a [helpful article](https://css-tricks.com/accessible-svgs/#aa-example-2-standalone-icon-decorative) on this subject. 

```suggestion
				return <SVGIcon aria-hidden="true" name="apple-logo" size="24" icon={ Apple } />;
			case 'Linux i686':
			case 'Linux i686 on x86_64':
			case LINUX_TAR_LINK:
			case LINUX_DEB_LINK:
				return <SVGIcon aria-hidden="true" name="linux-logo" size="24" icon={ Linux } />;
			case WINDOWS_LINK:
			default:
				return <SVGIcon aria-hidden="true" name="windows-logo" size="24" icon={ Windows } />;
```

File: client/blocks/get-apps/mobile-download-card.jsx [hunk 4]
```diff
@@ -162,16 +169,84 @@ class MobileDownloadCard extends Component {
...
+				/>
+			</div>
+		);
+	}
+
+	getQrCode() {
+		return (
+			<div className="get-apps__qr-code-subpanel jetpack">
+				<QRCode
+					value={ localizeUrl( 'https://apps.wordpress.com/mobile/?campaign=calypso-qrcode-apps' ) }
```
Comment (dcalhoun) [inline client/blocks/get-apps/mobile-download-card.jsx]: My understanding is the intention is for the QR code to direct users to the respective platform store: `https://wp.com/app`. I am uncertain as to whether the `campaign` is supported/tracked with that URL.

File: client/blocks/get-apps/desktop-download-card.jsx [hunk 5]
```diff
@@ -153,22 +229,122 @@ class DesktopDownloadCard extends Component {
...
+	}
+
+	getJetpackBrandedPanel() {
+		const { isMobile } = userAgent;
+
+		return (
+			<>
+				<div className="get-apps__card-text jetpack">
+					<SVGIcon
+						name="desktop-app-logo"
```
Comment (dcalhoun) [inline client/blocks/get-apps/desktop-download-card.jsx]: Similar to my other comment, I think it makes sense to hide this from screen readers. 
```suggestion
						aria-hidden="true"
						name="desktop-app-logo"
```

File: client/blocks/get-apps/mobile-download-card.jsx [hunk 6]
```diff
@@ -162,16 +169,84 @@ class MobileDownloadCard extends Component {
...
+		const isIos = isiPad || isiPod || isiPhone;
+
+		return (
+			<div className="get-apps__badges jetpack">
+				<p className="get-apps__card-text jetpack">
+					{ translate(
+						'Everything you need to publish, manage, and grow your site anywhere, any time.'
+					) }
+				</p>
+				<AppsBadge
```
Comment (dcalhoun) [inline client/blocks/get-apps/mobile-download-card.jsx:183]: It would appear I overlooked updating the `title` for the `AppBadge` component in previous PRs. 😬 Sorry about that. Would you be willing to add a conditional text referencing "Jetpack?" It currently always says the "WordPress <iOS|Android> mobile app." 

https://github.com/Automattic/wp-calypso/blob/2053b74187dbef03cb69a9dd1fd7cbe3b34e8629/client/blocks/get-apps/apps-badge.jsx#L28

https://github.com/Automattic/wp-calypso/blob/2053b74187dbef03cb69a9dd1fd7cbe3b34e8629/client/blocks/get-apps/apps-badge.jsx#L49

File: client/blocks/get-apps/style.scss [hunk 7]
```diff
@@ -7,27 +17,140 @@
...
+
+		&::before {
+			content: "";
+			background-color: #e9eff5;
+			position: absolute;
+			height: 100%;
+			width: 200vw;
+			left: -100vw;
+			z-index: -1;
+		}
```
Comment (dcalhoun) [inline client/blocks/get-apps/style.scss]: This is quite clever — I mean that in a positive way. That said, whenever we are able to remove the branding conditionals, it would likely be sensible to refactor the layout padding to make this absolutely positioned element unnecessary.

**Review by dcalhoun — COMMENTED** [review]
This looks really great! I did not notice any issues in comparison to the original design and it functioned really well in macOS Safari, both small and large screens. 

I left a few thoughts to consider, but none are blocking.

File: client/blocks/get-apps/desktop-download-card.jsx (hunk 1 above)
Comment (SiobhyB) [reply client/blocks/get-apps/desktop-download-card.jsx:92]: I appreciate your perspective here. Zooming out, I can see how the mutation would be hard to parse. I've gone ahead to create separate objects in https://github.com/Automattic/wp-calypso/pull/69674/commits/59edabb39eac9a83792917b4e052316b30742863, as per your suggestion.

File: client/blocks/get-apps/desktop-download-card.jsx (hunk 2 above)
Comment (SiobhyB) [reply client/blocks/get-apps/desktop-download-card.jsx]: Good catch! Fixed in https://github.com/Automattic/wp-calypso/pull/69674/commits/738f704fcc518ca7926dfd517b50c4ea14aa9ac4.

File: client/blocks/get-apps/mobile-download-card.jsx (hunk 4 above)
Comment (SiobhyB) [reply client/blocks/get-apps/mobile-download-card.jsx]: Aha, you're right. The current URL is just redirecting to the static apps.wordpress.com site on mobile, rather than the app stores. I also didn't find any results when searching our accounts for the campaign name, so your suspicion that's not supported with this current format seems right. 

I've gone ahead to update the URL (for both the WordPress and Jetpack panels) in https://github.com/Automattic/wp-calypso/pull/69674/commits/ed05e141e9689686876d499ec5041b53a82be1db. I used the `apps.wordpress.com/get?campaign=` structure that we're using in other places, which I believe is what we want.

In addition, this comment made me realise that I should have likely associated the new text link with a campaign. I've done that in https://github.com/Automattic/wp-calypso/pull/69674/commits/ecf94df2373ba79ceb4a8361a1b3a13ddaee2cf0.

Let me know if you think any other changes are needed to this and thanks for spotting this!

File: client/blocks/get-apps/desktop-download-card.jsx (hunk 5 above)
Comment (SiobhyB) [reply client/blocks/get-apps/desktop-download-card.jsx]: Implemented in https://github.com/Automattic/wp-calypso/pull/69674/commits/ddb0f0c17772d96967669a39cedc2eba2d06a89e, thank you!

File: client/blocks/get-apps/mobile-download-card.jsx (hunk 6 above)
Comment (SiobhyB) [reply client/blocks/get-apps/mobile-download-card.jsx:183]: Ah, good spot, I'd overlooked that too! Implemented that change in https://github.com/Automattic/wp-calypso/pull/69674/commits/91bdd1f2fb107b14837384399b9ad3602187b2d0.

File: client/blocks/get-apps/style.scss (hunk 7 above)
Comment (SiobhyB) [reply client/blocks/get-apps/style.scss]: Thank you, I agree it'd be nice to refactor so that everything is cleaner when we go about removing the branding conditionals.

**Comment by SiobhyB** [comment]
@dcalhoun, thank you for the review! I believe I've addressed all your points, but let me know if you spot anything else. 🙇‍♀️

File: client/blocks/get-apps/desktop-download-card.jsx [hunk 8]
```diff
@@ -53,32 +61,77 @@ class DesktopDownloadCard extends Component {
...
+
+							firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
+							secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
+							thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
+					  }
+					: {
+							firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
+							secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
+							thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
+					  };
```
Comment (dcalhoun) [inline client/blocks/get-apps/desktop-download-card.jsx]: @SiobhyB after reviewing [your latest change](https://github.com/Automattic/wp-calypso/pull/69674#discussion_r1012058709) in https://github.com/Automattic/wp-calypso/commit/59edabb39eac9a83792917b4e052316b30742863, I realize I completely misinterpreted the original code. I thought the same keys were being replaced. I realize now that we are conditionally adding keys. I think your original implementation makes total sense. Please feel free to revert https://github.com/Automattic/wp-calypso/commit/59edabb39eac9a83792917b4e052316b30742863 if you'd like. Sorry for the misguidance.

File: client/blocks/get-apps/apps-badge.jsx [hunk 9]
```diff
@@ -25,7 +25,10 @@ const APP_STORE_BADGE_URLS = {
 				: '335703880'; // WordPress
 			return `https://apps.apple.com/app/apple-store/id${ appId }?pt=299112&ct=${ utm_source }&mt=8`;
 		},
-		getTitleText: () => translate( 'Download the WordPress iOS mobile app.' ),
+		getTitleText: () =>
+			this.displayJetpackAppBranding
```
Comment (dcalhoun) [inline client/blocks/get-apps/apps-badge.jsx]: This line and its sibling for Android throws an error currently. Because of the structure of this configuration object, we must pass the branding flag as an argument to the function. We cannot reference a class property as done here. 

```
TypeError: undefined is not an object (evaluating 'undefined.displayJetpackAppBranding')
```

<details><summary>Patch to fix error</summary>

```diff
diff --git a/client/blocks/get-apps/apps-badge.jsx b/client/blocks/get-apps/apps-badge.jsx
index 225a473a9b..7946fd00bc 100644
--- a/client/blocks/get-apps/apps-badge.jsx
+++ b/client/blocks/get-apps/apps-badge.jsx
@@ -25,8 +25,8 @@ const APP_STORE_BADGE_URLS = {
 				: '335703880'; // WordPress
 			return `https://apps.apple.com/app/apple-store/id${ appId }?pt=299112&ct=${ utm_source }&mt=8`;
 		},
-		getTitleText: () =>
-			this.displayJetpackAppBranding
+		getTitleText: ( displayJetpackAppBranding ) =>
+			displayJetpackAppBranding
 				? translate( 'Download the Jetpack iOS mobile app.' )
 				: translate( 'Download the WordPress iOS mobile app.' ),
 		getAltText: () => translate( 'Apple App Store download badge' ),
@@ -49,8 +49,8 @@ const APP_STORE_BADGE_URLS = {
 			const appId = displayJetpackAppBranding ? 'com.jetpack.android' : 'org.wordpress.android';
 			return `https://play.google.com/store/apps/details?id=${ appId }&referrer=utm_source%3D%${ utm_source }%26utm_medium%3D${ utm_medium }%26utm_campaign%3D${ utm_campaign }`;
 		},
-		getTitleText: () =>
-			this.displayJetpackAppBranding
+		getTitleText: ( displayJetpackAppBranding ) =>
+			displayJetpackAppBranding
 				? translate( 'Download the Jetpack Android mobile app.' )
 				: translate( 'Download the WordPress Android mobile app.' ),
 		getAltText: () => translate( 'Google Play Store download badge' ),
@@ -157,7 +157,7 @@ export class AppsBadge extends PureComponent {
 				>
 					<img
 						src={ imageSrc }
-						title={ titleText ? titleText : badge.getTitleText() }
+						title={ titleText ? titleText : badge.getTitleText( this.displayJetpackAppBranding ) }
 						alt={ altText ? altText : badge.getAltText() }
 					/>
 				</a>

```

</details>

Comment (SiobhyB) [reply client/blocks/get-apps/apps-badge.jsx]: Oof, not sure how I managed to miss that 🤦‍♀️

Thank you for spotting so quickly! I've gone ahead to fix now.

**Review by dcalhoun — APPROVED** [review]

**Comment by paulforgione** [comment]
Hey @SiobhyB – this is looking great!

**1**. One thing I noticed is the light blue background color doesn't continue down the page if the browser window is particularly tall:

Height

**2.** I also noticed in your screenshots and on my device that the Lottie logo animation appears low resolution / blurry in Chrome. Firefox is nice and crisp though. I'm not sure if this is a browser-specific issue that we have any control over?

**3.** And lastly, the WordPress logo in the Desktop section appears to have 1 pixel being cropped off of the bottom?:
Logo-cropped


Regarding you questions:

> In the desktop app section, I used the same icon across both desktop and mobile devices, though the designs show different icons. Should I change this to use the different version of the WordPress icon for desktop (as per the design)?

The same icon for both desktop and mobile is fine, I realize now that's the icon for the desktop app, so that works!

> I know there was some discussion related to the mobile app section's copy. Should this be changed?

I'll check in on this to see where we landed.

> The font-sizes don't all precisely match up to the designs. In cases where the font-size in the designs didn't have a matching variable from this list, I went for the closest match:

Everything looks good to me, thanks for adapting. I'll make sure to reference these numbers in the future. :)

**Comment by SiobhyB** [comment]
Thanks so much @paulforgione! I've addressed your points below.

> 1. One thing I noticed is the light blue background color doesn't continue down the page if the browser window is particularly tall:

I've gone ahead to fix this, which you should be able to verify on the latest live branch. Let me know if any other issues come up related to the background color following the fix.

> 2. I also noticed in your screenshots and on my device that the Lottie logo animation appears low resolution / blurry in Chrome. Firefox is nice and crisp though. I'm not sure if this is a browser-specific issue that we have any control over?

I found that the blurriness was being caused by [the `translate3d()` CSS function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/translate3d) and that this has also been reported to Lottie's GitHub repo in https://github.com/airbnb/lottie-web/issues/2718. As the blurriness was very noticeable with this larger image, especially on larger monitors, I went ahead to implement the solution offered in that GitHub issue. There's some performance loss associated with this, but I think it's a worthwhile trade-off for the higher quality look of the image. I'd appreciate hearing if you have further thoughts on this, though.

| Before  | After |
| ------------- | ------------- |
|  |  |

> 3. And lastly, the WordPress logo in the Desktop section appears to have 1 pixel being cropped off of the bottom?:

I appreciate your eagle eyes here! 🙇‍♀️ I've gone ahead to fix this now.

I think the copy is the last part that needs addressing here, but let me know if you spot any other issues too. Thanks again for your help!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7625734

Thank you @SiobhyB for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7625734).
